src/Tizen.System/PowerUsage/PowerUsage.cs

```diff
@@ -0,0 +1,224 @@
+
+        /// <summary>
+        /// Display.
+        /// </summary>
+        Display = 3,
+
+        /// <summary>
+        /// Device Network.
+        /// </summary>
+        DeviceNetwork = 4,
+
+        /// <summary>
+        /// GPS Sensor.
+        /// </summary>
+        Gps = 5
+    }
+
+    /// <summary>
+    /// Provides information related to the power consumption by applications or by hardware resources on a battery-powered device for a given duration of time.
+    /// </summary>
+    /// <feature>http://tizen.org/feature/battery</feature>
+    /// <privilege>http://tizen.org/privilege/systemmonitor</privilege>
+    /// <since_tizen> 7 </since_tizen>
+    public static class PowerUsage
+    {
+        static PowerUsage()
+        {
+        }
+
+        private static double GetPowerUsage(IntPtr BMHandle, ResourceType rtype)
+        {
+            double batteryUsage = 0;
+            int ret = Interop.PowerUsage.GetPowerUsage(BMHandle, rtype, out batteryUsage);
+            if (((PowerUsageError)ret != PowerUsageError.None))
+            {
+                Log.Error(Globals.LogTag, "Error getting battery usage for resource" + (PowerUsageError)ret);
+                throw PowerUsageErrorFactory.ThrowPowerUsageException(ret);
+            }
+            return batteryUsage;
+        }
+
+        private static long ConvertDateTimeToTimestamp(DateTime dateTime)
+        {
+            DateTime epoch = new DateTime(1970, 1, 1, 0, 0, 0, 0, DateTimeKind.Utc);
+            return (long)((dateTime.ToUniversalTime() - epoch).TotalSeconds);
+        }
+
+        /// <summary>
+        /// Gets the battery consumption in mAh(milli-Ampere hour) for the resources specified by the application in custom interval.
+        /// </summary>
+        /// <since_tizen> 7 </since_tizen>
+        /// <param name="appID">Application ID of the application for which battery usage is required.</param>
+        /// <param name="start">Start Time for data in DateTime.</param>
+        /// <param name="end">End Time for data in DateTime.</param>
+        /// <param name="rtypes">list of resource type identifiers like BLE, WiFi, CPU etc.</param>
+        /// <returns>Returns the battery consumption in mAh(milli-Ampere hour) for the resources specified by the application in custom interval.</returns>
+        /// <feature>http://tizen.org/feature/battery</feature>
+        /// <privilege>http://tizen.org/privilege/systemmonitor</privilege>
+        /// <exception cref="ArgumentException">When an invalid parameter value is set.</exception>
+        /// <exception cref="UnauthorizedAccessException">If the privilege is not set.</exception>
+        /// <exception cref="InvalidOperationException">In case of any system error.</exception>
+        /// <exception cref="NotSupportedException">In case power usage is not supported</exception>
+        public static IList<double> GetPowerUsage(string appID, IList<ResourceType> rtypes, DateTime start, DateTime end)
```

Comment:
Please consider more better approach. It is not good idea to return a list as result. User should remember the index of rtypes to find any result of the specific resource type. struct or dictionary is more better choice you can use.

Reply:
Actually, this API returns the battery usage of all the resource types requested by user.
It is not necessary that user requires the battery usage of all the resources mentioned in enum Tizen.System.ResourceType.
The input is dynamic, so we are taking it in IList<ResourceType>.
The output is also dynamic corresponding to the input..
so we used IList<Double> as return type. 
Do you have any other suggestion for that ?

Reply:
He suggest to use dictionary
`Dictionary<ResourceType, double>`